csrc/transform_replay.h

```diff
@@ -136,19 +136,26 @@ class TORCH_CUDA_CU_API TransformReplay {
   // replay_resize indicates whether resize should be replayed or
   // ignored. It is only replayed when replaying a producer for
   // indexing.
+  // replay_allocation indicates whether to replace the producer's allocation
+  // domain with corresponding consumer's allocation domain. By default, we
+  // should preserve producer's current allocation domain, and if that
+  // allocation domain is inconsistent with the replay, an error will be raised.
+  // This option is used in cacheBefore, cacheAfter, and cacheFork
   static std::pair<TensorDomain*, size_t> replayPasC(
       const TensorView* producer,
       const TensorView* consumer,
       int64_t consumer_compute_at_axis,
       bool replay_swizzle = false,
-      bool replay_resize = false);
+      bool replay_resize = false,
+      bool replay_allocation = false);
```

Comment:
Oh no, yet another boolean option 😢 

Seriously, we should consider clean this up and make them more explicit. Not a blocker for this PR.

Reply:
I think we can have a
```C++
struct TransformReplayOptions {
  bool replay_swizzle = false;
  bool replay_resize = false;
  bool replay_allocation = false;
};
```
Does this make sense?

Reply:
I will clean it up in this PR. It's not much work.

Reply:
cleaned up